components/automate-cli/pkg/verifyserver/services/startmockserverservice/startmockserverservice.go

```diff
@@ -0,0 +1,160 @@
+package startmockserverservice
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+)
+
+// StartMockServerService provides functionality to start mock servers.
+type StartMockServerService struct{}
+
+// StartMockServer starts a mock server of the given type and port.
+func (servers *StartMockServerService) StartMockServer(cfg models.StartMockServerRequestBody) (*models.Server, error) {
+	switch cfg.Protocol {
+	case "tcp":
+		return servers.StartTCPServer(cfg.Port)
+	case "udp":
+		return servers.StartUDPServer(cfg.Port)
+	case "https":
+		return servers.StartHTTPSServer(cfg.Port, cfg.Cert, cfg.Key)
+	default:
+		return nil, errors.New("unsupported protocol")
+	}
+}
+
+func (s *StartMockServerService) StartTCPServer(port int) (*models.Server, error) {
+	// create a TCP listener on the specified port and
+	// save the listener instance in the handler struct
+	listener, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+
+	if err != nil {
+		return nil, errors.New(err.Error())
+	}
+	// defer listener.Close()
+
+	log.Printf("TCP server started on port %d", port)
```

Comment:
You can use logger declared in server.go by passing to service similar to what's done status api.